interface/patient_file/pos_checkout_normal.php

```diff
@@ -580,7 +580,7 @@ function markTaxes($taxrates)
             if ($GLOBALS['discount_by_money']) {
                 $amount  = sprintf('%01.2f', trim($_POST['form_discount']));
             } else {
-                $amount  = sprintf('%01.2f', trim($_POST['form_discount']) * $form_amount / 100);
+                $amount  = sprintf('%01.2f', trim($_POST['form_discount']) * $form_amount / (100 - (trim($_POST['form_discount']))));
```

Comment:
I strongly recommend pre processing any math so you can coalesce arguments. Consider type casting also.